db/migrations/2022_11_08_000000_create_angel_types_table.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Engelsystem\Migrations;
+
+use Engelsystem\Database\Migration\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use stdClass;
+
+class CreateAngelTypesTable extends Migration
+{
+    use ChangesReferences;
+
+    /**
+     * Creates the new table, copies the data and drops the old one
+     */
+    public function up(): void
+    {
+        $connection = $this->schema->getConnection();
+        $this->schema->create('angel_types', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('name')->unique();
+            $table->text('description')->default('');
+
+            $table->string('contact_name')->default('');
+            $table->string('contact_dect')->default('');
+            $table->string('contact_email')->default('');
+
+            $table->boolean('restricted')->default(false);
+            $table->boolean('requires_driver_license')->default(false);
+            $table->boolean('no_self_signup')->default(false);
+            $table->boolean('show_on_dashboard')->default(true);
+            $table->boolean('hide_register')->default(false);
+        });
+
+        if (!$this->schema->hasTable('AngelTypes')) {
+            return;
+        }
+
+        /** @var stdClass[] $records */
```

Comment:
I think this returns an Illuminate Collection or something like that.

Reply:
It does (and that contains the stdClass instances)